### PR TITLE
Feat: Bump Go version to v1.17 and upgrade the dependency k8s libs

### PR DESCRIPTION
**Pull request by wonderflow**
Signed-off-by: Jianbo Sun <jianbo.sjb@alibaba-inc.com>

### Description of your changes

Fixes #

I have:

- [ ] Read and followed KubeVela's [contribution process](https://github.com/oam-dev/kubevela/blob/master/contribute/create-pull-request.md).
- [ ] [Related Docs](https://github.com/oam-dev/kubevela.io) updated properly. In a new feature or configuration option, an update to the documentation is necessary. 
- [ ] Run `make reviewable` to ensure this PR is ready for review.
- [ ] Added `backport release-x.y` labels to auto-backport this PR if necessary.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -45,49 +44,232 @@ require (
...
 	gotest.tools v2.2.0+incompatible
-	helm.sh/helm/v3 v3.6.1
+	helm.sh/helm/v3 v3.8.0
 	istio.io/client-go v0.0.0-20210128182905-ee2edd059e02
-	k8s.io/api v0.22.1
-	k8s.io/apiextensions-apiserver v0.22.1
-	k8s.io/apimachinery v0.22.1
-	k8s.io/cli-runtime v0.21.0
-	k8s.io/client-go v0.22.1
+	k8s.io/api v0.23.3
```
Comment (barnettZQG) [inline go.mod:68]: Is it safe to change core dependencies in smaller versions?

Comment (wonderflow) [reply go.mod:68]: it's a dependency of helm v3

Comment (barnettZQG) [reply go.mod:68]: we can only upgrade the github.com/docker/cli package?

Comment (wonderflow) [reply go.mod:68]: The package ` github.com/docker/cli` depends by helm v3

Comment (barnettZQG) [reply go.mod:68]: replace github.com/docker/cli => github.com/docker/cli v20.10.9 ?

**Comment by wonderflow** [comment]
closing as https://github.com/kubevela/kubevela/pull/3776 merged

**Pull request closed without merging**